bin/gwd/gwd.ml

```diff
@@ -1930,53 +1882,87 @@ let main () =
     Util.cnt_dir := Secure.base_dir ();
   Wserver.stop_server :=
     List.fold_left Filename.concat !(Util.cnt_dir) ["cnt"; "STOP_SERVER"];
-  let (query, cgi) =
-    try Sys.getenv "QUERY_STRING", true with Not_found -> "", !force_cgi
-  in
-  if cgi then
+  let gtw = try Sys.getenv "GATEWAY_INTERFACE" with Not_found -> "" in
+  let query = try Sys.getenv "QUERY_STRING" with Not_found -> "" in
+  if gtw <> "" || !Wserver.cgi then
     begin
-      Wserver.cgi := true;
-      let is_post =
-        try Sys.getenv "REQUEST_METHOD" = "POST" with Not_found -> false
-      in
+      let is_post = try Sys.getenv "REQUEST_METHOD" = "POST" with Not_found -> false in
       let query =
         if is_post then
-          let len =
-            try int_of_string (Sys.getenv "CONTENT_LENGTH") with
-              Not_found -> -1
-          in
+          let len = try int_of_string (Sys.getenv "CONTENT_LENGTH") with Not_found -> -1 in
           set_binary_mode_in stdin true; read_input len
         else query
       in
       let addr =
         try Sys.getenv "REMOTE_HOST" with
           Not_found -> try Sys.getenv "REMOTE_ADDR" with Not_found -> ""
       in
-      let script =
-        try Sys.getenv "SCRIPT_NAME" with Not_found -> Sys.argv.(0)
-      in
-      geneweb_cgi addr (Filename.basename script) query
+      Wserver.noproc:= true;
+      match gtw with
+        | "RELAY/HTTP" -> 
+          Wserver.cgi:= false;
+          let path = try Sys.getenv "PATH_INFO" with Not_found -> "" in
+          geneweb_request addr path query
+        | _ ->
+          Wserver.cgi:= true;
+          let script = try Sys.getenv "SCRIPT_NAME" with Not_found -> Sys.argv.(0) in
+          geneweb_request addr (Filename.basename script) query
     end
   else geneweb_server ()
 
 let () =
+#ifdef DEBUG
+  Printexc.record_backtrace true;
+  Sys.enable_runtime_warnings false;
```

Comment:
Did you mean true? I picked (and corrected) these two lines here : https://github.com/geneweb/geneweb/commit/53d561ed13087cf1af52b15c5c8bf62dedb64d2f